File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/visitor/hosting/ComputeService.java
package com.bigboxcode.designpattern.visitor.hosting;

public class ComputeService implements Service {

    private final double price = 10.50;
    private int quantity;

    public ComputeService(int quantity) {
        this.quantity = quantity;
    }

    public double getPrice() {
        return price;
    }

    public int getQuantity() {
        return quantity;
    }

    @Override
    public double accept(HostingCalculatorVisitor hostingCalculatorVisitor) {
        return hostingCalculatorVisitor.visit(this);
    }
}

File: design-patterns/php/visitor/hostingcost/ComputeService.php
<?php
// ComputeService.php

namespace BigBoxCode\DesignPattern\Visitor\HostingCost;

class ComputeService implements Service {
    private const PRICE = 10.50;
    private int $quantity;

    public function __construct(int $quantity) {
        $this->quantity = $quantity;
    }

    public function getPrice(): float {
        return self::PRICE;
    }

    public function getQuantity(): int {
        return $this->quantity;
    }

    public function accept(HostingCalculatorVisitor $hostingCalculatorVisitor): float {
        return $hostingCalculatorVisitor->visitComputeService($this);
    }
}